controllers/dependencies.go

```diff
@@ -97,6 +97,18 @@ func (r *OperatorPipelineReconciler) applyManifests(fileName string, Namespace s
 	}
 
 	obj.SetNamespace(Namespace)
+
+	// Setting owner reference on created resources to
+	// allow for background deletion
+	ownerRefs := []metav1.OwnerReference{{
+		APIVersion: r.OperatorPipeline.APIVersion,
+		Kind:       r.OperatorPipeline.Kind,
+		Name:       r.OperatorPipeline.Name,
+		UID:        r.OperatorPipeline.UID,
+	}}
+
+	obj.SetOwnerReferences(ownerRefs)
```

Comment:
I believe this method should be leveraged to set the controller as the owner:
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference

Reply:
Interesting. I'm not sure about that. We can both use the controllerutil or the api machinery one that you see on the code. The struct for ownerRefs has other two optional fields being one a boolean that allows for setting the controller as owner. My question is: wouldn't the controller own the primary resource? Being the secondary ones owned by the primary? Like in the code? It actually works pretty well when the CR is deleted. But I didn't test the other one. I'll check that out. And come back here.

Reply:
yep, SetOwnerReference set the `BlockOwnerDeletion` to false, but SetControllerReference set it true. Here is the description of the field: https://github.com/kubernetes/apimachinery/blob/19377c9f105d03037e842e151d4605fef730b073/pkg/apis/meta/v1/types.go#L329

Reply:
Also, the controllerutil.SetOwnerReference does this for you. You don't have to handle setting all of the metav1.OwnerReference fields.

Reply:
@samira-barouti Is correct we need to use the method linked. We _also_ need to refactor from top town for this controller to this dependecy file to pass in the `OperatorPipeline` struct and not the `meta` struct. SetControllerReference throws an exception when `meta` is passed in. I was testing this extensively yesterday and it does not work otherwise.

Reply:
So in conclusion: 
1 - SetOwnerReference solves the problem of this PR but...
2 - SetControllerReference solves 2 problems:
  a. the background deletion
  b. mark the secondary resource to signal events
3 - The Owns function needs to work in conjunction with SetControllerReference. That is basically the controller saying you generate events and I'm listening. So if you delete a single pipeline or tasks it also triggers the controller and it reapplies all the resources again. How useful is this in this case is a different discussion.

That said all changes are implemented now.